app/scripts/components/common/chart/tooltip.tsx

```diff
@@ -38,23 +38,25 @@ const TooltipItem = styled(ListItem)`
 `;
 
 export default function TooltipComponent(props: TooltipComponentProps) {
-  const { colors, dateFormat, xKey, uniqueKeys, active, payload, label } =
+  const { colors, dateFormat, uniqueKeys, active, payload, label } =
     props;
+  
   const inactiveKeys = uniqueKeys.filter((e) => !e.active).map((e) => e.label);
   if (active && payload && payload.length) {
     return (
       <TooltipWrapper>
         <div>
-          <strong>{dateFormatter(label, dateFormat)}</strong>
+          <strong>{timeFormatter(label, dateFormat)}</strong>
         </div>
-        {Object.keys(payload[0].payload)
-          .filter((key) => key !== xKey && !inactiveKeys.includes(key))
-          .map((key, idx) => {
-            const point = payload[0].payload[key];
+        {uniqueKeys
+          .map((e,idx) => ({label: e.label, color: colors[idx]}))
+          .filter((key) =>!inactiveKeys.includes(key.label))
+          .map((key) => {
```

Comment:
Maybe, but I couldn't think how since I will lose the index of the key after filtering. All of this made me think It is better to bind the colors to unique key. so I moved the logic to higher level (in analysis/index)